File: 724-FindPivotIndex/724-FindPivotIndex.py
# Last updated: 8/13/2025, 8:19:11 PM
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        runningSum = [None]*len(nums)
        runningSum[0] = nums[0]

        for i in range(1, len(nums)) : 
            runningSum[i] = runningSum[i-1] + nums[i]

        for i in range(len(nums)) : 
            if i == 0 : 
                leftSum = 0 
                rightSum = runningSum[len(nums)-1] - runningSum[i]
            elif i == len(nums)-1 :
                leftSum = runningSum[i-1] 
                rightSum = 0
            else : 
                leftSum = runningSum[i-1]
                rightSum = runningSum[len(nums)-1] - runningSum[i]
            print(leftSum , rightSum)
            if leftSum == rightSum : 
                return i
        return -1

File: 724-FindPivotIndex/724-FindPivotIndex.java
// Last updated: 8/13/2025, 8:19:19 PM
class Solution {
    public int pivotIndex(int[] nums) {
        int n = nums.length;
        for(int i=1;i<n;i++){
            nums[i] += nums[i-1];
        }

        for(int i=0;i<n;i++){
            int left;
            int right;
            if(i==0){
                left =0;
            }
            else{
                left = nums[i-1];
            }
            if(i==n-1){
                right=0;
            }
            else{
                right = nums[n-1] - nums[i];
            }
            if(left==right){
                return i;
            }
        }
        return -1;
    }
}